backend/src/models/user.ts

```diff
@@ -11,12 +11,17 @@ class User extends Model<InferAttributes<User>, InferCreationAttributes<User>> {
   declare createdAt: CreationOptional<Date>;
   declare updatedAt: CreationOptional<Date>;
 
-  isValidPassword(password: string) {
-    return bcrypt.compare(password, this.password);
+  async isValidPassword(password: string) {
+    return await bcrypt.compare(password, this.password);
+  }
+
+  async encryptPassword(password: string) {
+    const hash = await bcrypt.hash(password, parseInt(process.env.SALT_ROUNDS || '10'));
+    return hash;
   }
```

Comment:
Nitpick: Couldn't this just be:
```js
encryptPassword(password: string) {
  return bcrypt.hash(password, parseInt(process.env.SALT_ROUNDS || '10'));
}
```

Reply:
Read your mind before review and changed that 😂